rust/abacus-core/src/types/provider_data.rs

```diff
@@ -0,0 +1,33 @@
+use ethers::prelude::{H256, U256};
+
+/// Info about a given block in the chain.
+#[derive(Debug)]
+pub struct BlockInfo {
+    /// Hash of this block
+    pub hash: H256,
+    /// Unix timestamp of this block's creation in seconds
+    pub timestamp: u64,
+    /// Block height or the nth block in the chain
+    pub number: u64,
+    /// Total used gas by transactions in this block
+    pub gas_used: U256,
+    /// Maximum amount of gas allowed in this block
+    pub gas_limit: U256,
+}
+
+/// Information about a given transaction in the chain.
+#[derive(Debug)]
+pub struct TxnInfo {
+    /// Hash of this transaction
+    pub hash: H256,
+    /// Amount of gas which was used by this transaction
+    pub gas_used: U256,
+    /// Price paid for gas on this txn.
+    pub gas_price: U256,
```

Comment:
Should we support EIP-1559 here?

Reply:
If you want the data in the database, it is easy enough to add it. But I was only extracting the data I thought we might want to poke into at this point. Again, easy to add if you think it would add value to the explorer or for our debugging later.

Reply:
Went ahead and added